sdk/apimanagement/api-management-custom-widgets-tools/src/node/deploy.ts

```diff
@@ -0,0 +1,80 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { CONFIG_FILE_NAME } from "../paths";
+import CustomWidgetBlobService from "./CustomWidgetBlobService";
+import fs from "fs";
+import getStorageSasUrl from "./getStorageSasUrl";
+import readdir from "./readdir";
+
+/**
+ * resourceId - resource ID of API Management service "subscriptions/[subscription-id]/resourceGroups/[resource-group-name]/providers/Microsoft.ApiManagement/service/[service-name]"
+ * managementApiEndpoint - URL with protocol (e.g. https://management.azure.com)
+ * apiVersion - optional to override default (e.g. "2019-01-01")
+ * tokenOverride - optional, provides token to use for auth, instead of 'az login' approach
+ */
+export type TServiceInformation = {
+  resourceId: string;
+  managementApiEndpoint: string;
+  apiVersion?: string;
+  tokenOverride?: string;
+};
+
+/**
+ * Deploys everything from /dist folder to the API Management DevPortals' blob storage.
+ *
+ * @param serviceInformation - service information for deployment
+ * @param name - name of the widget to be deployed
+ * @param fallbackConfigPath - local path to the config file (by default "./static/config.msapim.json")
+ */
+async function deploy(
+  serviceInformation: TServiceInformation,
+  name: string,
+  fallbackConfigPath = "./static/" + CONFIG_FILE_NAME
+): Promise<void> {
+  console.log("\n\n");
+  console.log("Starting deploy process of custom widget: " + name);
+  console.log("Please, sign in to your Azure account when prompted\n");
+
+  const blobStorageUrl = await getStorageSasUrl(serviceInformation);
+  const customWidgetBlobService = new CustomWidgetBlobService(blobStorageUrl, name);
+
+  let config;
+  try {
+    console.log("Looking for config file in the Azure blob storage");
+    config = await customWidgetBlobService.getConfig();
+  } catch (e) {
+    console.log("Config not found.");
+  }
+  if (!config) {
+    console.log("Looking for a local config file in: " + fallbackConfigPath);
+    config = JSON.parse(fs.readFileSync(fallbackConfigPath).toString());
```

Comment:
Since you are in a async function already why not use the asynchronouse version of readFile from `import {readFile} from "fs/promises";`

Reply:
I'm not sure about the complexity, it would just be a matter of adding `await` before `readFile`. `deploy` is already async